src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -146,10 +146,47 @@ class DataCube {
         this.dimensions.splice(axis, 1) // Remove dimension information
     }
 
+    applyDimension(process, dimension, target_dimension, context) {
+        const data = this.data;
+        const axis = this.dimensions.findIndex(e => e.name === dimension);
+        const labels = this.dimensions[axis].labels;
+        const allCoords = this._iterateCoords(data.shape.slice(), [axis]) // get the generator, axis of the selected dimension is `null` (entire dimension is selected)
+        const targetDimensionLabels = [];
+
+        if (target_dimension) {
+            if (this.getDimensionByName(target_dimension)) {
+                throw new Error("Dimension `target_dimension` already exists and cannot replace dimension `dimension`.");
+            }
+
+            const dim = this.getDimensionByName(dimension);
+            dim.name = target_dimension;
+            dim.type = this.OTHER;
+
+            for (let i = 0; i < data.shape[axis]; i++) {
+                targetDimensionLabels.push(i);
+            }
+        }
+
+        for (let coord of allCoords) {
+            const dataToProcess = convert_to_1d_array(data.pick.apply(data, coord));
+            dataToProcess.labels = target_dimension ? targetDimensionLabels : labels;
+            this._setArrayAlongAxis(coord, axis, process({ data: dataToProcess, context }));
+        }
+    }
+
     getDataShape() {
         return this.data.shape;
     }
 
+    _setArrayAlongAxis(coord, axis, array) {
+        for (let i = 0; i < array.length; i++) {
+            const newCoord = coord.slice();
+            newCoord[axis] = i;
+
+            this.data.set(...newCoord, array[i]);
```

Comment:
> Spread operator is actually faster in this case (no need to allocate a new array)

Indeed, it's _significantly_ faster (19s vs 13s for me) running the benchmark. Thanks for that, I was a bit too paranoid regarding the spread operator without even thinking about it. Using this in the normal `apply` improves its performance greatly too, I'll add it in the Datacube cleanup PR.